playbook/app/pb_kits/playbook/pb_advanced_table/_advanced_table.tsx

```diff
@@ -253,10 +266,117 @@ const AdvancedTable = (props: AdvancedTableProps) => {
+        });
+      }
+
+      items.push({
+        type: 'row',
+        row: row,
+        id: row.id
+      });
+
+      const isExpandable = row.getIsExpanded();
+      const rowHasNoChildren = row.original?.children && !row.original.children.length ? true : false;
+      const isDataLoading = isExpandable && (inlineRowLoading && rowHasNoChildren) && 
+        (row.depth < columnDefinitions[0].cellAccessors?.length);
+
+      if (isDataLoading) {
+        items.push({
+          type: 'loading',
+          row: row,
+          id: `loading-${row.id}`
+        });
+      }
+    });
+
+    return items;
+  }, [
+    virtualizedRows,
+    tableRows.rows,
+    props.tableOptions?.subRowHeaders,
+    inlineRowLoading,
+    columnDefinitions
+  ]);
+
+  const tableWrapperStyle = virtualizedRows ? {
+    overflow: 'auto', // scrollable table container
+    position: 'relative', // needed for sticky header
+    height: '600px', // fixed height - adjust as needed
+  } : {}
+
+  // Configure virtualizer to work with our flattened items
+  const virtualizerConfig = useVirtualizer({
+    count: virtualizedRows ? flattenedItems.length : tableRows.rows.length,
+    getScrollElement: () => tableWrapperRef.current,
+    estimateSize: (index) => {
+      if (!virtualizedRows || flattenedItems.length === 0) return 50;
+
+      const item = flattenedItems[index];
+      if (!item) return 50; // Default row height
+
+      if (item.type === 'header') return 40; // Header height
+      if (item.type === 'loading') return 30; // Loading indicator height
+      return 50; // Regular row height
+    },
+    overscan: 5,
+    getItemKey: (index) => virtualizedRows ? flattenedItems[index]?.id || index : tableRows.rows[index]?.id || index,
+  })
+
+  const virtualizer = virtualizedRows ? virtualizerConfig : null
+
+  const fetchNextPage = useCallback(() => {
+    if (isFetching || totalFetched >= totalDBRowCount) return; // Stop if already fetching or no more data
+
+    setIsFetching(true);
+
+    // Simulate a network request delay
+    setTimeout(() => {
+      const nextChunk = fullData.slice(totalFetched, totalFetched + fetchSize);
+      setDataChunk(prev => [...prev, ...nextChunk]);
+      setTotalFetched(prev => prev + nextChunk.length);
+      setIsFetching(false);
+    }, 500);
+  }, [isFetching, totalFetched, totalDBRowCount, fullData, fetchSize]);
+
+  const fetchMoreOnBottomReached = useCallback((containerRefElement?: HTMLDivElement | null) => {
+    if (containerRefElement) {
+      const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+      // If user scrolls near bottom, fetch more data
+      if (scrollHeight - scrollTop - clientHeight < 500 && !isFetching && totalFetched < totalDBRowCount) {
+        fetchNextPage();
+      }
+    }
+  }, [fetchNextPage, isFetching, totalFetched, totalDBRowCount]);
+
+  useEffect(() => {
+    fetchMoreOnBottomReached(tableWrapperRef.current);
+  }, [fetchMoreOnBottomReached]);
```

Comment:
@skduncan @nidaqg This is just an idea. It might be worth removing this code from the main component. I know we would not reuse it, but it could clarify the code. I am concerned that this file size will make it even harder for future developers to understand.